src/LaravelMsGraphMailServiceProvider.php

```diff
@@ -25,9 +25,9 @@ public function boot(): void
     {
         $this->app->bind(MicrosoftGraphApiService::class, function () {
             //throw exceptions when config is missing
-            throw_unless(filled(config('mail.mailers.microsoft-graph.tenant_id')), ConfigurationMissing::tenantId());
-            throw_unless(filled(config('mail.mailers.microsoft-graph.client_id')), ConfigurationMissing::clientId());
-            throw_unless(filled(config('mail.mailers.microsoft-graph.client_secret')), ConfigurationMissing::clientSecret());
+            throw_if(blank(config('mail.mailers.microsoft-graph.tenant_id')), ConfigurationMissing::tenantId());
+            throw_if(blank(config('mail.mailers.microsoft-graph.client_id')), ConfigurationMissing::clientId());
+            throw_if(blank(config('mail.mailers.microsoft-graph.client_secret')), ConfigurationMissing::clientSecret());
 
             return new MicrosoftGraphApiService(
                 tenantId: config('mail.mailers.microsoft-graph.tenant_id', ''),
```

Comment:
With `level: max`, PHPStan is failing here:

```
 ------ --------------------------------------------------------------------------------------------------------------------------------------- 
  Line   LaravelMsGraphMailServiceProvider.php
 ------ ---------------------------------------------------------------------------------------------------------------------------------------
  33     Parameter $tenantId of class InnoGE\LaravelMsGraphMail\Services\MicrosoftGraphApiService constructor expects string, mixed given.
  34     Parameter $clientId of class InnoGE\LaravelMsGraphMail\Services\MicrosoftGraphApiService constructor expects string, mixed given.
  35     Parameter $clientSecret of class InnoGE\LaravelMsGraphMail\Services\MicrosoftGraphApiService constructor expects string, mixed given.
  36     Parameter $accessTokenTtl of class InnoGE\LaravelMsGraphMail\Services\MicrosoftGraphApiService constructor expects int, mixed given.
 ------ ---------------------------------------------------------------------------------------------------------------------------------------
```

While looking through this, I noticed that the way the config is retrieved forces the key/name of the mailer to literally be `microsoft-graph`. However, in large applications it could be useful to have different named mailers using the `microsoft-graph` driver. Thus, I would propose the following:

- get the configuration values from the `array $config` that is passed in line 40, similar to `$config['from']['address']`
- validate that the configuration values exist (throw `ConfigurationMissing` if not) and are of the correct type (throw `ConfigurationInvalid` if not)
- directly instantiate `MicrosoftGraphApiService` instead of getting it from the service container (also remove the binding)

Would you be ok with those changes?

Reply:
Thanks a lot for digging into this and suggesting improvements! I appreciate your effort. Your idea sounds useful, especially for large apps, but I'm worried it might lead to breaking changes and adding complexity. Could you open a PR with these changes? That way, we can take a closer look and discuss it further. Just a heads-up, though, I can't promise it'll be merged. Thanks again!